File: 백준/Silver/18110. solved．ac/solved．ac.java
import java.util.*;
import java.io.*;


public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(br.readLine());
        }
        Arrays.sort(arr);
        int exc = Math.round((float) (n * 15) / 100);

        int sum = 0;

        for (int i = exc; i < n - exc; i++) {
            sum += arr[i];
        }
        int answer = n == 0 ? 0 : Math.round((float) sum / (n - 2 * exc));

        System.out.println(answer);

    }

}

File: 백준/Silver/18110. solved．ac/solved．ac.py
import sys

input = sys.stdin.readline


def custom_round(x):
    return int(x + 0.5)


n = int(input().strip())
arr = [int(input().strip()) for _ in range(n)]

# 1. 정렬
arr.sort()

# 2. 상위 15%, 하위 15% 제외
exc = custom_round(n * 15 / 100)

sum = 0
for i in range(exc, n-exc):
    sum += arr[i]

answer = 0 if n == 0 else custom_round(sum / (n - 2*exc))

print(answer)
